src/stapi_fastapi/routers/product_router.py

```diff
@@ -279,7 +279,7 @@ async def search_opportunities_sync(
             case Failure(e) if isinstance(e, ConstraintsException):
                 raise e
             case Failure(e):
-                logger.error(
+                logger.exception(
```

Comment:
these shouldn't use logger.exception because they're not within an `except` context -- they should just be `logger.error` -- see https://docs.python.org/3/library/logging.html#logging.Logger.exception

```
logger.exception("An error occurred while searching opportunities")
```

The

Reply:
Good stuff. Thanks.